Source/KeyboardManager.swift

```diff
@@ -5,109 +5,160 @@
+        eventClosure?(.willShow(data))
+        innerEventClosure?(.willShow(data))
     }
 
     @objc
     private func keyboardDidShow(_ notification: Notification) {
-        eventClosure?(.didShow, extractData(from: notification))
+        let data = extractData(from: notification)
+        eventClosure?(.didShow(data))
+        innerEventClosure?(.didShow(data))
     }
 
     @objc
     private func keyboardWillHide(_ notification: Notification) {
-        eventClosure?(.willHide, extractData(from: notification))
+        let data = extractData(from: notification)
+        eventClosure?(.willHide(data))
+        innerEventClosure?(.willHide(data))
     }
 
     @objc
     private func keyboardDidHide(_ notification: Notification) {
-        eventClosure?(.didHide, extractData(from: notification))
+        let data = extractData(from: notification)
+        eventClosure?(.didHide(data))
+        innerEventClosure?(.didHide(data))
     }
 
-    private func extractData(from notification: Notification) -> KeyboardManagerEventData {
+    private func extractData(from notification: Notification) -> KeyboardManagerEvent.Data {
         guard let endFrame = notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue,
-              let beginFrame = notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue,
-              let curve = notification.userInfo?[UIKeyboardAnimationCurveUserInfoKey] as? String,
-              let isLocal = notification.userInfo?[UIKeyboardIsLocalUserInfoKey] as? NSNumber,
-              let duration = notification.userInfo?[UIKeyboardAnimationDurationUserInfoKey] as? NSNumber else {
-            return KeyboardManagerEventData.null()
+            let beginFrame = notification.userInfo?[UIKeyboardFrameBeginUserInfoKey] as? NSValue,
+            let curve = notification.userInfo?[UIKeyboardAnimationCurveUserInfoKey] as? NSNumber,
+            let isLocal = notification.userInfo?[UIKeyboardIsLocalUserInfoKey] as? NSNumber,
+            let duration = notification.userInfo?[UIKeyboardAnimationDurationUserInfoKey] as? NSNumber else {
+            return KeyboardManagerEvent.Data.null()
         }
-        let frame = KeyboardManagerEventData.Frame(begin: beginFrame.cgRectValue, end: endFrame.cgRectValue)
-        return KeyboardManagerEventData(
-                frame: frame,
-                animationCurve: curve,
-                animationDuration: duration.doubleValue,
-                isLocal: isLocal.boolValue
+        let frame = KeyboardManagerEvent.Frame(begin: beginFrame.cgRectValue, end: endFrame.cgRectValue)
+        return KeyboardManagerEvent.Data(
+            frame: frame,
+            animationCurve: curve.intValue,
+            animationDuration: duration.doubleValue,
+            isLocal: isLocal.boolValue
         )
     }
-
 }
 
 extension KeyboardManager: KeyboardManagerProtocol {
+    public func bindToKeyboardNotifications(scrollView: UIScrollView) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/alphatroya/KeyboardManager/blob/25a2ec034b913c48e066e9db453abead1303374d/Source/KeyboardManager.swift#L139"></span>Public declarations should be documented.</td>
    </tr>
  </tbody>
</table>